resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -94,6 +94,14 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private var totalCoresAcquired = 0
   private var totalGpusAcquired = 0
 
+  // The amount of time to wait for locality scheduling
+  private val localityWait = conf.getTimeAsMs("spark.locality.wait", "3s")
```

Comment:
I added `spark.locality.wait` as a `ConfigEntry`. Please take a look to see if I did that correctly.